### DISCUSSION TITLE
quarto convert: support an option to retain empty code cells

### DISCUSSION THREAD

**Discussion by anandology** [quarto-dev/quarto-cli#11685] [Feature Requests]
### Description

The `quarto convert` command when converting from `qmd` to `ipynb`, drops all the code cells which are empty. This may be fine in most use cases, there are cases where this is undesirable. 

It would be nice to have an option to retain empty code cells as well.

## Sample Use Case 

I teach programming courses and I use `quarto convert` to generate `ipynb files`  from `qmd` sources. These notebooks are used by the students of the course. The following is  an example where empty cells are needed.

    The following code creates a new function `square`.

    ```{python}
    def square(x):
        return x*x
    ```

    ```{python}
    square(4)
    ````

    Can you try creating a new function `cube` to compute cube a of number 
    in the following cell?

    ```{python}

    ```

    Check if your `cube` function is working correctly by running the following examples.

    ```{python}
    print(cube(2))
    print(cube(3))
    print(cube(4))
    ```

## The issue

The third code cell in the above example gets dropped. 

The work-around is to make the code cell not empty by adding a comment, but I would prefer to keep it empty.

## The Code

The part of the code that drops empty cells is at:

https://github.com/quarto-dev/quarto-cli/blob/main/src/core/jupyter/jupyter.ts#L422

**Comment by cscheid** [comment]
This seems like a fine request to me - I don't truly understand why we're dropping empty code cells here. I'll open an issue to track.

**Reply by mcanouil** [reply]
- https://github.com/quarto-dev/quarto-cli/issues/11690

**Reply by anandology** [reply]
Thanks @cscheid and @mcanouil.
